README.md

```diff
@@ -1 +1,155 @@
+- **Liquibase**: Controls database changes, ensuring consistency across environments.
+- **Jackson**: Handles efficient JSON parsing and generation.
+- **Swagger**: Generates interactive API documentation.
+- **MySQL**: Manages and stores application data as a relational database system.
+- **Docker**: Develops, ships, and runs applications within containers for consistent environments.
+
+
+## Layered Architecture
+
+
+- **Controller Layer**: Manages incoming HTTP requests and client communications.
+- **Service Layer**: Contains business logic, processing data, and interacting with repositories.
+- **Repository Layer**: Handles direct database interactions for querying and managing data.
+- **Security Layer**: Manages authentication and authorisation processes.
+- **DTO Layer**: Transfers data between application layers, including only necessary information.
+- **Mapper Layer**: Converts between object models to ensure data consistency.
+- **Test Layer**: Validates functionality through unit and integration tests.
+
+
+## Database Entities
+
+
+- **Book**: Represents a book with metadata such as title, author, and publication year.
+- **Category**: Classifies books into genres or subjects.
+- **Shopping Cart**: Temporarily holds selected products for potential purchase.
+- **Cart Item**: Specifies a book and its quantity in the shopping cart.
+- **User**: Registered individual with personal details and assigned permissions.
+- **Role**: Defines user permissions, such as USER or ADMIN.
+- **Order**: Records completed transactions, capturing purchase details.
+- **Order Item**: Details specific books purchased in an order.
+
+
+## Access to Endpoints
+
+
+### User
+- 🟩 **POST** `/auth/registration`: User registration.
+- 🟩 **POST** `/auth/login`: User login.
+
+
+### Book
+- 🟨 **GET** `/books`: Retrieve book catalogue.
+- 🟨 **GET** `/books/{id}`: View book details.
+- 🟨 **GET** `/books/search`: Search books.
+- 🟥 **POST** `/books`: Add a new book.
+- 🟥 **PUT** `/books/{id}`: Update book details.
+- 🟥 **DELETE** `/books/{id}`: Delete book.
+
+
+### Category
+- 🟨 **GET** `/categories`: Retrieve categories.
+- 🟨 **GET** `/categories/{id}`: View category.
+- 🟨 **GET** `/categories/{id}/books`: Get books by category.
+- 🟥 **POST** `/categories`: Create a category.
+- 🟥 **PUT** `/categories/{id}`: Update category.
+- 🟥 **DELETE** `/categories/{id}`: Delete category.
+
+
+### Shopping Cart
+- 🟨 **GET** `/cart`: View cart.
+- 🟨 **POST** `/cart`: Add to cart.
+- 🟨 **PUT** `/cart/cart-items/{id}`: Update cart.
+- 🟨 **DELETE** `/cart/cart-items/{id}`: Remove from cart.
+
+
+### Order
+- 🟨 **GET** `/orders`: View orders history.
+- 🟨 **GET** `/orders/{id}/items`: View order items.
+- 🟨 **GET** `/orders/{orderId}/items/{itemId}`: View specific order item.
+- 🟨 **POST** `/orders`: Place order.
+- 🟥 **PATCH** `/orders/{id}`: Update order status.
+
+
+## SQL Database Diagram
+
+
+![Database Diagram]
```

Comment:
you should push the image to the git as well(some "readme resources" folder in the root of the project) to be able to see it on the GitHub and not only on your local env